index.bs

```diff
@@ -368,6 +370,160 @@ dictionary SanitizerConfig {
 };
 </pre>
 
+## Configuration Invariants ## {#invariants}
+
+An empty configuration allows everything (when called with the "unsafe" methods).
+A configuration `"default"` contains a [=built-in safe default configuration=]. Note
+that different sanitizer methods have different defaults.
+
+Configurations can and should be modified by developers to suit their
+purposes. Options are to write a new [[#config|configuration dictionary]] from
+scratch, to modify an existing {{Sanitizer}}'s configuration by using the
+modifier methods, or to {{Sanitizer/get()}} an existing {{Sanitizer}}'s
+[[#config|configuration]]  as a dictionary and modify the dictionary and then
+create a new {{Sanitizer}} with it.
+
+Several conditions need to hold for a configuration to be valid:
+
+- Mixing global allow- and remove-lists:
+  - {{SanitizerConfig/elements}} or {{SanitizerConfig/removeElements}} may exist,
+    but not both.
+    If both are missing, this is equivalent to {{SanitizerConfig/removeElements}}
+    [=map/set=] to &laquo; [] &raquo;.
+  - {{SanitizerConfig/attributes}} or {{SanitizerConfig/removeAttributes}} may exist,
+    but not both.
+    If both are missing, this is equivalent to {{SanitizerConfig/removeAttributes}}
+    [=map/set=] to &laquo; [] &raquo;.
+  - {{SanitizerConfig/dataAttributes}} is conceptually an extension of the
+    {{SanitizerConfig/attributes}} allow-list. This attribute is only allowed when the
+    corresponing list is allowed.
+- Duplicate entries between different global lists:
+  - There are no duplicate entries (i.e. no same elements) between
+    {{SanitizerConfig/elements}}, {{SanitizerConfig/removeElements}}, or
+    {{SanitizerConfig/replaceWithChildrenElements}}.
+  - There are no duplicate entries (i.e. no same attributes) between
+    {{SanitizerConfig/attributes}} or {{SanitizerConfig/removeAttributes}}.
+
+
+The {{SanitizerConfig/elements}} element allow-list can also specify allowing or removing
+attributes for a given element. This is meant to mirror [[HTML]]'s structure, which knows
+both [=/global attributes=] as well as local <a spec=dom>attributes</a> that apply to a
+specific element. For {{SanitizerConfig|Sanitizer configurations}}, it is assumed that
+configuration authors would prefer either all-global or all-local attribute declarations.
```

Comment:
Why is that assumed? Even the default config has both "local" and "global" attributes.